src/performance/metrics/StatisticCollector.cpp

```diff
@@ -317,6 +319,884 @@ double StatisticCollector::getLoadAverage() {
+            for (int i = 0; i < 8; i++) {
+                long long value = 0;
+                if (*p && (*p >= '0' && *p <= '9')) {
+                    value = strtoll(p, &p, 10);
+                    while (*p == ' ') p++;
+                }
+                cpuStats.push_back(value);
+            }
+            firstReading.push_back(cpuStats);
+        } else {
+            break;  // No more CPU lines
+        }
+    }
+    fclose(file1);
+    
+    // Sleep for a short interval to get meaningful difference
+    usleep(100000);  // 100ms
+    
+    // Second reading
+    std::vector<std::vector<long long>> secondReading;
+    FILE *file2 = fopen("/proc/stat", "r");
+    if (!file2) {
+        stat_logger.error("Cannot open /proc/stat for second reading");
+        return cpuUsages;
+    }
+    
+    // Skip the first line (total cpu stats)
+    fgets(line, sizeof(line), file2);
+    
+    // Read per-CPU stats
+    while (fgets(line, sizeof(line), file2) != NULL) {
+        if (strncmp(line, "cpu", 3) == 0 && line[3] >= '0' && line[3] <= '9') {
+            std::vector<long long> cpuStats;
+            char *p = line;
+            // Skip "cpu" and cpu number
+            while (*p && *p != ' ') p++;
+            while (*p == ' ') p++;
+            
+            // Parse CPU time values
+            for (int i = 0; i < 8; i++) {
+                long long value = 0;
+                if (*p && (*p >= '0' && *p <= '9')) {
+                    value = strtoll(p, &p, 10);
+                    while (*p == ' ') p++;
+                }
+                cpuStats.push_back(value);
+            }
+            secondReading.push_back(cpuStats);
+        } else {
+            break;  // No more CPU lines
+        }
+    }
+    fclose(file2);
+    
+    // Calculate usage for each CPU
+    size_t numCpus = std::min(firstReading.size(), secondReading.size());
+    for (size_t i = 0; i < numCpus; i++) {
+        if (firstReading[i].size() >= 4 && secondReading[i].size() >= 4) {
+            // Calculate total time difference
+            long long totalDiff = 0;
+            long long idleDiff = 0;
+            
+            for (int j = 0; j < 8 && j < (int)firstReading[i].size() && j < (int)secondReading[i].size(); j++) {
+                long long diff = secondReading[i][j] - firstReading[i][j];
+                totalDiff += diff;
+                if (j == 3) {  // idle time is the 4th field (index 3)
+                    idleDiff = diff;
+                }
+            }
+            
+            // Calculate CPU usage percentage
+            double usage = 0.0;
+            if (totalDiff > 0) {
+                usage = ((double)(totalDiff - idleDiff) / totalDiff) * 100.0;
+                if (usage < 0.0) usage = 0.0;
+                if (usage > 100.0) usage = 100.0;
+            }
+            cpuUsages.push_back(usage);
+        }
+    }
+    
+    return cpuUsages;
+}
+
+double StatisticCollector::getProcessSwitchesPerSecond() {
```

Comment:
I think here also we have capacity of moving common code to a function and call it in this segment